module/validation/receipt_validator_test.go

```diff
@@ -745,3 +752,154 @@ func (s *ReceiptValidationSuite) TestValidateReceiptAfterBootstrap() {
 	err := s.receiptValidator.ValidatePayload(candidate)
 	s.Require().NoError(err)
 }
+
+// TestValidateReceiptResultWithoutReceipt tests a case when a malicious leader incorporates a made-up execution result
+// into their proposal. ReceiptValidator must ensure that for each result included in the block, there must be
+// at least one receipt included in that block as well.
+func (s *ReceiptValidationSuite) TestValidateReceiptResultWithoutReceipt() {
+	// assuming signatures are all good
+	s.publicKey.On("Verify", mock.Anything, mock.Anything, mock.Anything).Return(true, nil)
+
+	// G <- A <- B
+	blocks, result0, seal := unittest.ChainFixture(2)
+	s.SealsIndex[blocks[0].ID()] = seal
+
+	receipts := unittest.ReceiptChainFor(blocks, result0)
+	blockA, blockB := blocks[1], blocks[2]
+	receiptA, receiptB := receipts[1], receipts[2]
+
+	blockA.Payload.Receipts = []*flow.ExecutionReceiptMeta{}
+	blockB.Payload.Receipts = []*flow.ExecutionReceiptMeta{receiptA.Meta()}
+	blockB.Payload.Results = []*flow.ExecutionResult{&receiptA.ExecutionResult}
+	// update block header so that blocks are chained together
+	unittest.ReconnectBlocksAndReceipts(blocks, receipts)
+	// assuming all receipts are executed by the correct executor
+	for _, r := range receipts {
+		r.ExecutorID = s.ExeID
+	}
+
+	for _, b := range blocks {
+		s.Extend(b)
+	}
+	s.PersistedResults[result0.ID()] = result0
+
+	candidate := unittest.BlockWithParentFixture(blockB.Header)
+	candidate.Payload = &flow.Payload{
+		Receipts: []*flow.ExecutionReceiptMeta{},
+		Results:  []*flow.ExecutionResult{&receiptB.ExecutionResult},
+	}
+
+	err := s.receiptValidator.ValidatePayload(candidate)
+	s.Require().Error(err)
+	s.Require().True(engine.IsInvalidInputError(err))
+}
+
+// TestValidateReceiptResultHasNotEnoughReceipts tests a case when a malicious leader incorporates an execution result
```

Comment:
I think we can remove this test case. If `k` is constant, then this is equivalent to the previous test.